packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -118,16 +119,25 @@ class CodeSizeController {
     changeDiffRoot(_activeDiffRoot);
   }
 
-  void loadTreeFromJsonFile(DevToolsJsonFile jsonFile) {
-    changeSnapshotJsonFile(jsonFile);
+  void loadTreeFromJsonFile(
+    DevToolsJsonFile jsonFile,
+    void Function(String error) onError,
+  ) {
 
     Map<String, dynamic> processedJson;
     if (jsonFile.isApkFile) {
       // APK analysis json should be processed already.
       processedJson = jsonFile.data;
     } else {
-      processedJson = treemapFromJson(jsonFile.data);
+      try {
+        processedJson = treemapFromJson(jsonFile.data);
+      } catch (error) {
+        onError('Failed to load snapshot: File type not supported.');
```

Comment:
let's add some more details:

"Failed to load snapshot: file type not supported.

The code size tool supports Dart AOT v8 snapshots, instruction sizes, and apk-analysis files. See documentation for how to generate these files."

Reply:
Done